### PR TITLE
Move the GitBook root and publish only the content needed

### PR DESCRIPTION
**Pull request by MikkCZ**
I have noticed that the `gh-pages` branch contains more than needed (scripts, encrypted key and all config files), potentially everything what ends up in the repository, or in the GitBook build folder. But the only files we actually want to publish are the documents - the rest should stay in `master` only.

This change also gives us freedom to have different information for the GitBook readme page and for the repository readme.

Thoughts @mozilla-l10n/pm-group ?

TODO:
- [x] verify automation
- [ ] remote links directly to files in GH repo
- [x] "styleguide" vs "style guide"
- [ ] **squash before/during merge**

### CODE REVIEW

**Comment by flodolo** [comment]
This definitely looks interesting. Were you able to test the automation on your fork?

To be completely honest, I'm not concerned about publishing more than necessary on GitHub pages.

Having a separate README is definitely nice, one concern is about existing link to styleguides in GitHub. It also invalidates existing pull requests.

I defer to the English speakers for "styleguide" vs "style guide". I assume we have this issue in a lot of places.

**Comment by MikkCZ** [comment]
> This definitely looks interesting. Were you able to test the automation on your fork?

I haven't tested the deploy script. The thing I changed the directory where `gitbook` looks for the sources when run in the repository root, but the output directory and output there is the same (excluding the non-content files). `gitbook serve` looks good. I will give the automation a deeper look to verify.

> Having a separate README is definitely nice, one concern is about existing link to styleguides in GitHub. It also invalidates existing pull requests.

If the links are relative in the markdown, nothing should break. To be honest I am not sure if markdown even support any other kind of links.
For PRs, they would need to be updated, especially if they create a completely new folder, as there would be no conflict. Or we can defer merging this after the other PRs are resolved in reasonable time.

**Comment by MikkCZ** [comment]
I have tested the deploy script and it works. The changes I have made are only to make the behaviour more standard with the gitbook defaults.

Btw. I see it resets my `.gitconfig`. I will update it to not do that again if someone runs it locally. ;)

**Comment by MikkCZ** [comment]
> I meant link from the outside to a specific styleguide in GitHub instead of GitBook, like https://github.com/mozilla-l10n/styleguides/blob/master/cs/README.md

Hmm, yes, you are right. These would stop working and there is no way to fix it from out side. But I would expect everyone or most people links to GitBook, not the repository. But I may be wrong of course.

**Comment by flodolo** [comment]
> Hmm, yes, you are right. These would stop working and there is no way to fix it from out side. But I would expect everyone or most people links to GitBook, not the repository. But I may be wrong of course.

We had the same issue with our documentation. Unfortunately, there's nothing you can do to fix that.
I've checked Pontoon and it links to GitBook, so we're good there.

File: docs/README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Style Guides
+
+This repository is used to store localization style guides for Mozilla communities. All style guides and related documents are listed in the menu on the left of the page.
+
+## Updating the documentation
+
+We use a [Github repository](https://github.com/mozilla-l10n/styleguides/) to store all documents. Simply open a pull request adding the new file or updating an existing document. Make sure to follow [these style guidelines](https://github.com/mozilla-l10n/documentation/blob/master/misc/documentation_styleguide.md), and note that all pull requests need to be reviewed before merging. The instruction for creating  new community style guides are part of the [guidelines for community style guides](guidelines/styleguide_guidelines.md).
```
Comment (flodolo) [inline docs/README.md]: Github -> GitHub
instruction for -> instructions for
also remove double space

Comment (MikkCZ) [reply docs/README.md]: Thank you, fixed now.

**Comment by MikkCZ** [comment]
As for other PRs, at this moment the most "critical" is #82, which won't report any conflict.

**Comment by flodolo** [comment]
Asking also for #84

**Comment by flodolo** [comment]
Can you rebase? If it works fine on your fork, I'm happy to give it a try and merge this ;-)

**Comment by flodolo** [comment]
> I defer to the English speakers for "styleguide" vs "style guide". I assume we have this issue in a lot of places.

@gueroJeff 
Any take on this: styleguide vs style guide?

**Comment by MikkCZ** [comment]
I have rebased and moved the new pt-BR files to the right place, so we should squash the commits before/during merge.

**Comment by flodolo** [comment]
I'll go ahead and make a call about "style guide"
https://en.wikipedia.org/wiki/Style_guide

**Pull request merged**